src/LinuxTracing/UprobesUnwindingVisitorTest.cpp

```diff
@@ -0,0 +1,211 @@
+  MOCK_METHOD(void, OnAddressInfo, (orbit_grpc_protos::FullAddressInfo), (override));
+  MOCK_METHOD(void, OnTracepointEvent, (orbit_grpc_protos::FullTracepointEvent), (override));
+  MOCK_METHOD(void, OnModulesSnapshot, (orbit_grpc_protos::ModulesSnapshot), (override));
+  MOCK_METHOD(void, OnModuleUpdate, (orbit_grpc_protos::ModuleUpdateEvent), (override));
+};
+
+class UprobesUnwindingVisitorTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    visitor_ = std::make_unique<UprobesUnwindingVisitor>(&maps_, &unwinder_);
+    visitor_->SetListener(&listener_);
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kUprobesMapsStart), Lt(kUprobesMapsEnd))))
+        .WillRepeatedly(Return(&kUprobesMapInfo));
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kTargetMapsStart), Lt(kTargetMapsEnd))))
+        .WillRepeatedly(Return(&kTargetMapInfo));
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kNonExecutableMapsStart), Lt(kNonExecutableMapsEnd))))
+        .WillRepeatedly(Return(&non_executable_map_info_));
+  }
+
+  void TearDown() override { visitor_.reset(); }
+
+  MockLibunwindstackMaps maps_;
+  MockLibunwindstackUnwinder unwinder_;
+  MockTracerListener listener_;
+
+  std::unique_ptr<UprobesUnwindingVisitor> visitor_ = nullptr;
+
+  static constexpr uint64_t kUprobesMapsStart = 42;
+  static constexpr uint64_t kUprobesMapsEnd = 84;
+
+  static constexpr uint64_t kTargetMapsStart = 100;
+  static constexpr uint64_t kTargetMapsEnd = 200;
+
+  static constexpr uint64_t kNonExecutableMapsStart = 500;
+  static constexpr uint64_t kNonExecutableMapsEnd = 600;
+
+  static constexpr uint64_t kTargetAddress1 = 100;
+  //  static constexpr uint64_t kUprobesAddress = 42;
+  static constexpr uint64_t kTargetAddress2 = 200;
+  static constexpr uint64_t kTargetAddress3 = 300;
+  //  static constexpr uint64_t kNonExecutableAddress3 = 500;
+
+  static inline const std::string kUprobesName = "[uprobes]";
+  static inline const std::string kTargetName = "target";
+  static inline const std::string kNonExecutableName = "data";
+
+  static inline unwindstack::MapInfo kUprobesMapInfo{
+      nullptr, kUprobesMapsStart, kUprobesMapsEnd, 0, PROT_EXEC | PROT_READ, kUprobesName};
+
+  static inline unwindstack::MapInfo kTargetMapInfo{nullptr, kTargetMapsStart,      kTargetMapsEnd,
+                                                    0,       PROT_EXEC | PROT_READ, kTargetName};
+
+  static inline unwindstack::MapInfo non_executable_map_info_{
+      nullptr, kNonExecutableMapsStart, kNonExecutableMapsEnd,
+      0,       PROT_EXEC | PROT_READ,   kNonExecutableName};
+};
+
+}  // namespace
+
+TEST_F(UprobesUnwindingVisitorTest, VisitValidStackSampleWithoutUprobesEmitsEvents) {
+  constexpr uint32_t kPid = 10;
+  constexpr uint64_t kStackSize = 13;
+  StackSamplePerfEvent event{kStackSize};
+  perf_event_sample_id_tid_time_streamid_cpu sample_id{
+      .pid = kPid, .tid = 11, .time = 15, .stream_id = 12, .cpu = 0, .res = 0};
```

Comment:
Just a personal preference and note that if you add a trailing comma after the last "parameter" then clang-format puts each in a different line, making it easier to read.

Reply:
Yes, looks better. Thanks.